### ISSUE TITLE
Add ffprobe

### ISSUE THREAD

**Issue by yakir12** [JuliaIO/VideoIO.jl#99]
How much more work would it be to add `ffprobe` functionality to this package? I personally need to get the date and time metadata from a bunch of videos.

**Comment by yakir12** [comment]
Allow me to rephrase: is the information for `original date and time` as well as `duration` accessible from a `VideoIO.AVInput`? If yes, could you point me to the right direction as to how to extract it? If I manage, I'll PR a function for these two queries.

**Comment by kmsquire** [comment]
See https://discourse.julialang.org/t/date-time-and-duration-from-a-video-file/5799/2

**Comment by kmsquire** [comment]
Copying from that post:

In code, this would look something like this:

```julia
v = VideoIO.testvideo("annie_oakley")
fc = unsafe_load(v.apFormatContext[1])
duration = fc.duration / VideoIO.AV_TIME_BASE
start_time_realtime = fc.start_time_realtime
no_start_time = (start_time_realtime == VideoIO.AV_NOPTS_VALUE)

println("Duration: $duration seconds")
if no_start_time
    println("Start time not available")
else
    println("start_time_realtime: $start_time_realtime")
end
```

Output:
```
Duration: 24.2242 seconds
Start time not available
```

**Comment by yakir12** [comment]
Awesome! I'll try and work in a PR for a clean API tomorrow. Thanks!

**Comment by yakir12** [comment]
Ah.. I forgot about the ffmpeg v 3.1 barrier. I'm suffering from it as well (I'm on ffmpeg version 3.2.7-1) and can't use VideoIO. It would be pretty impossible for me to build this API without actually using VideoIO. 

Should I downgrade ffmpeg? or do you think your fix #108 will get merged soonish?

**Comment by kmsquire** [comment]
Still working on #108. It passes locally, but getting it to pass Travis has been challenging. 

For your particular purpose, do you actually need to read the video images?  If not, then it should work to open the videos and get the info you need. (might need #98).

**Comment by yakir12** [comment]
Right you are. Right now I'm good with just that, but soon (like in a week) I'll need to fetch frames. Thanks a ton for all of this!

**Comment by kmsquire** [comment]
(But the easiest thing to do is downgrade to ffmpeg 2.8 or 2.9)

**Comment by yakir12** [comment]
It seems like the `start_time_realtime` field in `AVFormatContext` rarely contains any other value than the Unix epoch... Maybe a better place to look for that data is in the `metadata` field? Any hints on how to look in there?
Thanks!

**Comment by yakir12** [comment]
Wow, the speed is insane! I tried this out on a bunch of different videos and it's blazing fast compared to something like [exiftool](https://www.sno.phy.queensu.ca/~phil/exiftool/)!!!
But for all the videos I looked at, the `start_time_realtime` only shows `1970-01-01T00:00:00`...

**Comment by ianshmean** [comment]
This functionality was merged in #153 and if other `ffprobe` functionality is needed, the executable can accessed through:
```julia
using VideoIO
withenv(VideoIO.execenv) do
    out = VideoIO.collectexecoutput(`$(VideoIO.ffprobe) <<standard ffprobe args>>`)
end
```
